src/coreclr/tools/aot/ILCompiler.Compiler/Compiler/Dataflow/FlowAnnotations.cs

```diff
@@ -592,41 +592,18 @@ protected override TypeAnnotations CreateValueFromKey(TypeDesc key)
                         }
                     }
 
-                    if (IsAutoProperty(property))
+                    FieldDesc? backingField = backingFieldFromSetter ?? backingFieldFromGetter;
+                    if (backingField is not null)
                     {
-                        FieldDesc? backingField = null;
-                        if ((property.SetMethod is not null
-                                && property.SetMethod.HasCustomAttribute("System.Runtime.CompilerServices", "CompilerGeneratedAttribute")
-                                && backingFieldFromSetter is null)
-                            || (property.GetMethod is not null
-                                && property.GetMethod.HasCustomAttribute("System.Runtime.CompilerServices", "CompilerGeneratedAttribute")
-                                && backingFieldFromGetter is null))
+                        if (annotatedFields.Any(a => a.Field == backingField))
```

Comment:
Don't we want to know that the field annotations and property annotations are different? Is it a conflict if both are annotated the same way?